src/ripple/app/ledger/TxHold.cpp

```diff
@@ -0,0 +1,727 @@
+    : txn_(txn)
+    , lastRelayed_(0), txID_(txn->getTransactionID())
+    , account_(txn->getSourceAccount().getAccountID())
+    , sequence_(txn->getSequence())
+    , transactionParams_(params)
+{
+    feeLevel_ = txn->getFeeLevelPaid(loadBase, engine.getLedger()->getBaseFee());
+
+    if (txn->isFieldPresent(sfLastLedgerSequence))
+    {
+        lastValid_ = txn->getFieldU32(sfLastLedgerSequence);
+    }
+
+    if (txn->isFieldPresent(sfAccountTxnID))
+    {
+        priorTxID_ = txn->getFieldH256(sfAccountTxnID);
+    }
+}
+
+class TxQAccount
+{
+    friend class TxQImpl;
+
+protected:
+    //boost::intrusive::set_member_hook<> byAccount_;
+
+private:
+
+    Account                    account_;
+    //uint64_t                   feeLevel_;
+    uint64_t                   totalFees_;
+    // Sequence number will be used as the key.
+    std::map <uint32_t, CandidateTxn> transactions_;
+
+public:
+    TxQAccount(STTx::ref txn);
+    TxQAccount(const Account& account);
+
+    //int getFeeLevel() const
+    //{
+    //    return feeLevel_;
+    //}
+
+    int getTxnCount() const
+    {
+        return transactions_.size();
+    }
+
+    bool empty() const
+    {
+        return !getTxnCount();
+    }
+
+    // Returns true if this is the new head transaction
+    // for this account
+    CandidateTxn&  addCandidate (CandidateTxn&&);
+    bool removeCandidate(uint32_t const&);
+
+    std::pair<bool, CandidateTxn const*> findCandidateAt(uint32_t const&);
+};
+
+TxQAccount::TxQAccount(STTx::ref txn)
+    :TxQAccount(txn->getFieldAccount160(sfAccount))
+{
+}
+
+TxQAccount::TxQAccount(const Account& account)
+    : account_(account)
+    //, feeLevel_(0)
+    , totalFees_(0)
+{ 
+}
+
+CandidateTxn&
+TxQAccount::addCandidate(CandidateTxn&& txn)
+{
+    auto fee = txn.getFeeLevel();
+    totalFees_ += fee;
+    auto sequence = txn.getSequence();
+
+    transactions_.emplace(std::make_pair(sequence, std::move(txn)));
+    assert(&transactions_.at(sequence) != &txn);
+
+    return transactions_.at(sequence);
+}
+
+bool
+TxQAccount::removeCandidate(uint32_t const& sequence)
+{
+    return transactions_.erase(sequence) != 0;
+}
+
+std::pair<bool, CandidateTxn const*>
+TxQAccount::findCandidateAt(uint32_t const& sequence)
+{
+    auto iter = transactions_.find(sequence);
+    if (iter == transactions_.end())
+        return std::make_pair(false, nullptr);
+
+    return std::make_pair(true, &iter->second);
+}
+
+
+class GreaterFee
+{
+public:
+    bool operator() (const CandidateTxn& lhs, const CandidateTxn& rhs) const
+    {
+        return lhs.getFeeLevel() > rhs.getFeeLevel();
+    }
+};
+
+/*
+class LessAccount
+{
+public:
+    bool operator() (const TxQAccount& lhs, const TxQAccount& rhs) const
+    {
+        return lhs.getAccount() < rhs.getAccount();
+    }
+};
+
+class CompareAccount
+{
+public:
+    bool operator() (const Account& lhs, const Account& rhs) const
+    {
+        return lhs == rhs;
+    }
+};
+*/
+
+class TxQImpl : public TxQ
```

Comment:
In a blurry sort of way, I get the sense that `TxQ` provides a superset of the capabilities of the work done by the `LocalTx` class.  Would it be possible to remove `LocalTx` now that this is in place?  I'm hoping that one complicated holding tank for possibly failed transactions will be enough.  I prefer not to have two or three such holding tanks.